Network/Public/Include/Network/Connection.hpp

```diff
@@ -73,22 +70,21 @@ class Connection : public std::enable_shared_from_this<Connection>
     Message _messageBuffer;
 
     /**
-     * @brief Method for sending message header.
-     * @details Function asio::async_write is used to write the header of the message /
-     * to the socket. /
-     * If the writing header to the socket is successful and the message body isn't empty, /
-     * the body of the message is written (method writeBody()). If it is empty, /
-     * the current message is removed from message queue to send and next message header /
-     * from the message queue is sent. If the writing header to the socket failed, the error /
-     * message - "[connection id] Write Header Fail." is displayed.
-     */
+    * @brief Method for sending message header.
+    * @details Function asio::async_write is used to write the header of the message /
+    * to the socket. /
+    * If the writing header to the socket is successful and the message body isn't empty, /
+    * the body of the message is written (method writeBody()). If it is empty, /
+    * the current message is removed from message queue to send and next message header /
+    * from the message queue is sent. If the writing header to the socket failed, the error /
+    * message - "[connection id] Write Header Fail." is displayed.
+    */
     void writeHeader()
     {
         yas::shared_buffer bodyBuffer;
 
-        SerializationHandler handler;
-        handler.setNext(new CompressionHandler())->setNext(new EncryptionHandler());
-        MessageProcessingState result = handler.handleOutcomingMessage(_outcomingMessagesQueue.front(), bodyBuffer);
+        CombiningHandlers handler;
```

Comment:
You're using twice the same object in different methods, I'd like to move 'em to private scope of class. Before they were two separate objects and it wasn't so important